### PR TITLE
FIX:  fix ipfwd/test_mtu testcase failure in backend storage topology

### PR DESCRIPTION
**Pull request by yejianquan**
### Description of PR

Summary:
Fixes https://github.com/Azure/sonic-mgmt/issues/3879 

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Fix the nightly testcase failure.
https://github.com/Azure/sonic-mgmt/issues/3879 
#### How did you do it?
Update interface/sub-interface fetch logic to compatible with backend storage t1 topology, details in the code.

#### How did you verify/test it?
Run the testcase and make sure they passed
ipfwd/test_mtu.py::test_mtu[str2-7050qx-32s-acs-03-None-1514] PASSED [ 50%]
ipfwd/test_mtu.py::test_mtu[str2-7050qx-32s-acs-03-None-9114] PASSED [100%]

### CODE REVIEW

File: tests/common/devices/sonic.py [hunk 1]
```diff
@@ -1189,7 +1190,25 @@ def get_extended_minigraph_facts(self, tbinfo, namespace = DEFAULT_NAMESPACE):
...
+
+    # assert whether a topo is 'backend' type
+    def assert_topo_is_backend(self, tbinfo):
+        topo_key = constants.TOPO_KEY
+        name_key = constants.NAME_KEY
+        if topo_key in tbinfo.keys() and name_key in tbinfo[topo_key].keys():
+            topo_name = tbinfo[topo_key][name_key]
+            if constants.BACKEND_TOPOLOGY_FLAG in topo_name:
+                return True
+        else:
```
Comment (neethajohn) [inline tests/common/devices/sonic.py:1210]: you can get rid of this 'else' and just return False

File: tests/common/constants.py [hunk 2]
```diff
@@ -1,3 +1,9 @@
 VLAN_SUB_INTERFACE_SEPARATOR = "."
 # default port mapping mode for storage backend testbeds
 PTF_PORT_MAPPING_MODE_DEFAULT = "use_sub_interface"
+TOPO_KEY = "topo"
+NAME_KEY = "name"
+# field in mg_facts to flag whether it's a backend topology or not
+IS_BACKEND_TOPOLOGY_KEY = "is_backend_topology"
+# a topology whos name contains 'backend' will be considered as a backend topology
+BACKEND_TOPOLOGY_FLAG = "backend"
```
Comment (neethajohn) [inline tests/common/constants.py:9]: Rename this variable. 'FLAG' suffix is misleading. Generally the term 'flag' is to indicate a boolean value

File: tests/ipfwd/conftest.py [hunk 3]
```diff
@@ -45,15 +46,31 @@ def get_lag_facts(dut, lag_facts, switch_arptable, mg_facts, ignore_lags, enum_f
...
+
     selected_port_facts = {}
     up_port = None
     for a_intf_name, a_intf_data in port_status['int_status'].items():
         if a_intf_data['oper_state'] == 'up' and a_intf_name not in ignore_intfs:
             # Got a port that is up and not already used.
-            for intf in mg_facts['minigraph_interfaces']:
-                if intf['attachto'] == a_intf_name:
+            for intf in interfaces:
+
```
Comment (yxieca) [inline tests/ipfwd/conftest.py:65]: minor issue. please remove this empty line

File: tests/common/constants.py (hunk 2 above)
Comment (yejianquan) [reply tests/common/constants.py]: Done, currently use 'IND'(indicator) instead of 'FLAG'

File: tests/common/devices/sonic.py (hunk 1 above)
Comment (yejianquan) [reply tests/common/devices/sonic.py]: Done

File: tests/ipfwd/conftest.py (hunk 3 above)
Comment (yejianquan) [reply tests/ipfwd/conftest.py]: Done, removed the empty line

**Review by lolyu — APPROVED** [review]

**Review by neethajohn — APPROVED** [review]

**Pull request merged**